server/v2/server.go

```diff
@@ -30,27 +34,42 @@ type HasConfig interface {
 	Config() any
 }
 
-var _ ServerModule = (*Server)(nil)
+// HasStartFlags is a server module that has start flags.
+type HasStartFlags interface {
+	StartFlags() *pflag.FlagSet
+}
+
+var _ ServerModule[transaction.Tx] = (*Server)(nil)
 
 // Configs returns a viper instance of the config file
 func ReadConfig(configPath string) (*viper.Viper, error) {
 	v := viper.New()
-	v.SetConfigFile(configPath)
+
 	v.SetConfigType("toml")
+	v.SetConfigName("config")
+	v.AddConfigPath(configPath)
 	if err := v.ReadInConfig(); err != nil {
 		return nil, fmt.Errorf("failed to read config: %s: %w", configPath, err)
 	}
+
+	v.SetConfigType("toml")
+	v.SetConfigName("app")
+	v.AddConfigPath(configPath)
```

Comment:
That's duplicated right?

Reply:
yeah should only SetConfigName since set type & path before